cpp/src/io/csv/writer_impl.cu

```diff
@@ -19,17 +19,25 @@
  * @brief cuDF-IO CSV writer class implementation
  */
 
-#include "writer_impl.hpp"
+#include "durations.hpp"
+
+#include "csv_common.h"
+#include "csv_gpu.h"
```

Comment:
These should follow the same include behavior as the rest of `io` and use `#include <cudf/io/detail/******.hpp>`

Reply:
I agree, but these includes have just been moved up from `writer_impl.hpp`. I'm not sure what to do with `csv_common` and `csv_gpu` yet. Likely they'll be merged in to `writer_impl.cu`, but that's out of scope for this PR. Since these aren't new dependencies, can we ignore them for now?

Reply:
My mistake, I thought these files are in `cpp/include` and not `cpp/src`. The current pattern matches that of including private headers inside `cpp/src`